inc/saip4ext.h

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 20XX Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saip4ext.h
+ *
+ * @brief   This module defines SAI P4 Extension interface
+ */
+
+#if !defined (__SAIP4EXT_H_)
+#define __SAIP4EXT_H_
+
+#include <saitypes.h>
+
+/**
+ * @brief Attribute Id for P4 ext
+ */
+typedef enum _sai_p4ext_entry_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_P4EXT_ENTRY_ATTR_START,
+
+    /**
+     * @brief SAI P4 EXT table id
+     *
+     * @type sai_s8_list_t
```

Comment:
is this string table name ? where did it come frome?

Reply:
Yes this is the table name. Added an explanation for this in the HLD [here](https://github.com/opencomputeproject/SAI/pull/1293/files#diff-f10288ab3ba3d4cfacfc71aeeb21c72dfd6ed605f50c7f1db4ca10115c425c71R75-R79)

Reply:
what expected lengths are of this string? maybe it could fit to chardata[32] which is default in sai attribute?

Reply:
Table name might fit into the 32byte chardata however for other attributes of p4 extension objects we would still need an arbitrary length string as there can be variable number of action parameters or match fields. Using the same data type for all the attributes here keeps things more consistent. Also having arbitrary length for table name allows vendors to use fqn for the table names if they need to, which can be longer than 32 bytes

Reply:
ok, let leave it as is